### PR TITLE
Kafka Alerter and examples

### PR DESCRIPTION
**Pull request by lucasponce**
image

This PR adds a new Alerter to listen Apache Kafka topics and parse records as Data or Events.

Also an example is added to demo this new functionality.

### CODE REVIEW

File: alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+import org.hawkular.commons.properties.HawkularProperties;
+
+/**
+ * This is the main class of the Kafka Alerter.
+ *
+ * The Kafka Alerter will listen for triggers tagged with "Kafka" tag. The Alerter will create a consumer to a Kafka Topic
+ * with the config provider into the Trigger context. The Alerter will convert Kafka records into Hawkular Data or Events
+ * and send them into the Alerting engine.
+ *
+ * The Elasticsearch Alerter uses the following conventions for trigger tags and context:
```
Comment (jshaughn) [inline alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java]: cut+paste issue

File: alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+ *
+ * <pre>
+ *
+ * - [Required]    trigger.tags["Kafka"] = "<value reserved for future uses>"
```
Comment (jshaughn) [inline alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java]: Maybe "kafka" should be lowercase?

File: alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+@Alerter(name = "kafka")
+public class KafkaAlerter implements AlerterPlugin {
+    private static final MsgLogger log = MsgLogging.getMsgLogger(KafkaAlerter.class);
+
+    private static final String KAFKA_ALERTER = "hawkular-alerts.kafka-alerter";
+    private static final String KAFKA_ALERTER_ENV = "KAFKA_ALERTER";
+    private static final String KAFKA_ALERTER_DEFAULT = "true";
```
Comment (jshaughn) [inline alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java]: Our packaged alerters are enabled by default but as we add alerters I wonder if this is a mistake given the overhead of trigger processing, maybe alerters should be explicitly enabled.

File: alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java (hunk 1 above)
Comment (lucasponce) [inline alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java]: +1

File: alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java (hunk 2 above)
Comment (lucasponce) [inline alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java]: Per consistency, as Elasticsearch is first letter uppercase, I think Kafka makes sense to follow this.

File: alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java (hunk 3 above)
Comment (lucasponce) [inline alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java]: I think we commented some weeks ago that we can enable them by default.
That's good for community as less steps to test the project.
I think there is space for performance improvements and yes, at any sign that is creating some overhead we can disable it in the config.
As our goal is to impact in community in short term, I think is good to maintain them enabled (all of them, elasticsearch, prometheus, kafka and others).

File: alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java (hunk 2 above)
Comment (lucasponce) [inline alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java]: in any case, we can check with ignoreCase() in the code.

Comment (jshaughn) [inline alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java]: Actually, there is no consistency :)  Today we have "Elasticsearch" and "prometheus".  I didn't realize we had capitalized Elasticsearch.

Comment (lucasponce) [inline alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java]: oh, you are right :-) prometheus is lowercase.
Well, I would leave it as it is for now, perhaps we can modify the alerters tags in following iteration.
About Kafka most of documentation I've read to built this integration references it in first letter upper case.

**Comment by mperezco** [comment]
This is great @lucasponce 
It enables so many possibilities :-D

**Pull request merged**